File: try/prime.hs
import System.CPUTime
import Control.Monad

isPrime :: Int -> Bool
isPrime n | n <= 1 = False
          | otherwise = null [x | x <- [2..floor . sqrt $ fromIntegral n], n `mod` x == 0]

main :: IO ()
main = do
    start <- getCPUTime
    let c = length $ filter isPrime [0..9000000]
    print c
    end <- getCPUTime
    let diff = (fromIntegral (end - start)) / (10^12)
    printf "Computation time: %0.3f sec\n" (diff :: Double)

File: src/benchmarks/prime.nim
import math, times

proc is_prime(n: int): bool =
  if n <= 1:
    return false
  let endVal = int(sqrt(float(n)))
  for i in 2..endVal:
    if n mod i == 0:
      return false
  return true

let start = cpuTime()
var c = 0
for i in 0..<9000000:
  if is_prime(i):
    inc c

let endTime = cpuTime()
echo c
echo int((endTime - start) * 1000), "ms"
